staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -382,8 +382,8 @@ func (h *kmsv2PluginProbe) rotateDEKOnKeyIDChange(ctx context.Context, statusKey
 		if errGen == nil {
 			klog.V(6).InfoS("successfully rotated DEK",
 				"uid", uid,
-				"newKeyID", resp.KeyID,
-				"oldKeyID", state.KeyID,
+				"newKeyIDHash", fmt.Sprintf("%x", sha256.Sum256([]byte(resp.KeyID))),
```

Comment:
Can we use a `hashPool` similar to 
https://github.com/kubernetes/kubernetes/blob/7581ae812327fc8218204f678143a6f116cad931/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics.go#L283-L290

Reply:
@aramase I couldn't find the obvious location to initialize the hashPool in config.go. Even if we do we will need to pass it in `LoadEncryptionConfig` and keep propagating it wherever we need it. With all the other functionalities we have in config.go, this seems very trivial to me. So I am thinking of keeping this as is. wdyt?

Reply:
I do not have any strong opinion on using a hash pool, but we should maintain the `sha256:` prefix and empty key ID check:

https://github.com/kubernetes/kubernetes/blob/a6890b361de8d5bc7e90005dc2c5c167d9d80093/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics.go#L293-L297

Also, since there is quite a lot of logic in this log statement now, I would guard it with `if klog.V(6).Enabled() {` to avoid the computation since most servers do not run at such a high log level.